libs/openFrameworks/gl/ofGLProgrammableRenderer.cpp

```diff
@@ -1315,12 +1321,46 @@ void ofGLProgrammableRenderer::bind(const ofFbo & fbo, bool setupPerspective){
 }
 
 //----------------------------------------------------------
-void ofGLProgrammableRenderer::unbind(const ofFbo & fbo){
-	glBindFramebuffer(GL_FRAMEBUFFER, 0);
+void ofGLProgrammableRenderer::end(const ofFbo & fbo){
+	fbo.unbind();
```

Comment:
yeah, only talking about thread safety in the cpu, particularly about avoiding singletons (right now only the current window which contains a current renderer). not even about the renderer itself being thread safe, each thread should create it's own renderer

everything in the GL side is just responsibility of the user.

shareContextWith allows 2 windows usually in the same thread to share resources like textures or buffers but it's something that GLFW itself allows and again responsibility of the user to use correctly. 

you could even create 2 windows in different threads that share no GL objects at all but they can work without crashing by not accessing any singletons

sorry if it wasn't clear before